format/FormatCBFMiniPilatusDLS6MSN126.py

```diff
@@ -200,128 +195,6 @@ def _goniometer(self):
-        t0 = thickness
-
-        # FIXME would also be very nice to be able to take into account the
-        # misalignment of the individual modules given the calibration...
-
-        # single detector or multi-module detector
-
-        pixel_x *= 1000.0
-        pixel_y *= 1000.0
-        distance *= 1000.0
-
-        if not self._multi_panel:
-            detector = self._detector_factory.simple(
-                "PAD",
-                distance,
-                (beam_x * pixel_x, beam_y * pixel_y),
-                "+x",
-                "-y",
-                (pixel_x, pixel_y),
-                (nx, ny),
-                (underload, overload),
-                [],
-                ParallaxCorrectedPxMmStrategy(mu, t0),
-            )
-
-            for f0, f1, s0, s1 in determine_pilatus_mask(detector):
-                detector[0].add_mask(f0 - 1, s0 - 1, f1, s1)
-
-            detector[0].set_thickness(thickness)
-            detector[0].set_material("Si")
-            detector[0].set_mu(mu)
-
-            return detector
-
-        # got to here means 60-panel version
-        d = Detector()
-
-        beam_centre = matrix.col((beam_x * pixel_x, beam_y * pixel_y, 0))
-
-        fast = matrix.col((1.0, 0.0, 0.0))
-        slow = matrix.col((0.0, -1.0, 0.0))
-        s0 = matrix.col((0, 0, -1))
-        origin = (distance * s0) - (fast * beam_centre[0]) - (slow * beam_centre[1])
-
-        root = d.hierarchy()
-        root.set_local_frame(fast.elems, slow.elems, origin.elems)
-
-        xmins = [0, 494, 988, 1482, 1976]
-        xmaxes = [487, 981, 1475, 1969, 2463]
-        ymins = [0, 212, 424, 636, 848, 1060, 1272, 1484, 1696, 1908, 2120, 2332]
-        ymaxes = [195, 407, 619, 831, 1043, 1255, 1467, 1679, 1891, 2103, 2315, 2527]
-
-        self.coords = {}
-
-        fast = matrix.col((1.0, 0.0, 0.0))
-        slow = matrix.col((0.0, 1.0, 0.0))
-        panel_idx = 0
-        for ymin, ymax in zip(ymins, ymaxes):
-            for xmin, xmax in zip(xmins, xmaxes):
-                xmin_mm = xmin * pixel_x
-                ymin_mm = ymin * pixel_y
-
-                origin_panel = fast * xmin_mm + slow * ymin_mm
-
-                panel_name = "Panel%d" % panel_idx
-                panel_idx += 1
-
-                p = root.add_panel()
-                p.set_type("SENSOR_PAD")
-                p.set_px_mm_strategy(ParallaxCorrectedPxMmStrategy(mu, t0))
-                p.set_name(panel_name)
-                p.set_image_size((xmax - xmin, ymax - ymin))
-                p.set_trusted_range((underload, overload))
-                p.set_pixel_size((pixel_x, pixel_y))
-                p.set_thickness(thickness)
-                p.set_material("Si")
-                p.set_mu(mu)
-                p.set_local_frame(fast.elems, slow.elems, origin_panel.elems)
-                p.set_raw_image_offset((xmin, ymin))
-                self.coords[panel_name] = (xmin, ymin, xmax, ymax)
-
-        return d
-
     def _read_cbf_image(self):
```

Comment:
Presumably this is a duplicated function as well?

Looks like yup:

```
Grey-Area format :) [pilatus_multi_panel] $ grep _read_cbf_image *py
FormatCBFFullPilatus.py:    def _read_cbf_image(self):
FormatCBFFullPilatus.py:            data = self._read_cbf_image()
FormatCBFMini.py:    def _read_cbf_image(self):
FormatCBFMini.py:            data = self._read_cbf_image()
FormatCBFMiniADSCHF4M.py:    def _read_cbf_image(self):
FormatCBFMiniADSCHF4M.py:            data = self._read_cbf_image()
FormatCBFMiniEiger.py:    def _read_cbf_image(self):
FormatCBFMiniEiger.py:            data = self._read_cbf_image()
FormatCBFMiniPilatus.py:            raw_data = self._read_cbf_image()
FormatCBFMiniPilatusDLS12M.py:    def _read_cbf_image(self):
FormatCBFMiniPilatusDLS12M.py:            raw_data = self._read_cbf_image()
FormatCBFMiniPilatusDLS6MSN100.py:    def _read_cbf_image(self):
FormatCBFMiniPilatusDLS6MSN126.py:    def _read_cbf_image(self):
```

possible opportunity to reduce duplication further?

Reply:
Actually the ones in the superclass are a simplified version of the one in the base class - I don't know whether this is for a particular reason, or if using the one from the base class would work fine everywhere else...

https://github.com/cctbx/dxtbx/blob/0a251fb1533b043e409abed7d68e9eec3ea836be/format/FormatCBFMiniPilatusDLS6MSN100.py#L346-L362

https://github.com/cctbx/dxtbx/blob/0a251fb1533b043e409abed7d68e9eec3ea836be/format/FormatCBFMini.py#L231-L259